File: src/main/java/org/bk/graphql/config/DataLoaderRegistrationConfig.java
package org.bk.graphql.config;

import org.bk.graphql.domain.BookId;
import org.bk.graphql.web.AuthorsWrapper;
import org.bk.graphql.web.BookAuthorsDataLoader;
import org.springframework.context.annotation.Configuration;
import org.springframework.graphql.execution.BatchLoaderRegistry;

@Configuration
public class DataLoaderRegistrationConfig {
    private final BatchLoaderRegistry batchLoaderRegistry;
    private final BookAuthorsDataLoader bookAuthorsDataLoader;

    public DataLoaderRegistrationConfig(
        BatchLoaderRegistry batchLoaderRegistry,
        BookAuthorsDataLoader bookAuthorsDataLoader
    ) {
        this.batchLoaderRegistry = batchLoaderRegistry;
        this.bookAuthorsDataLoader = bookAuthorsDataLoader;
        
        batchLoaderRegistry.forTypePair(BookId.class, AuthorsWrapper.class)
            .registerMappedBatchLoader(bookAuthorsDataLoader);
    }
}

File: src/main/kotlin/org/bk/graphql/config/DataLoaderRegistrationConfig.kt
package org.bk.graphql.config

import org.bk.graphql.domain.BookId
import org.bk.graphql.web.AuthorsWrapper
import org.bk.graphql.web.BookAuthorsDataLoader
import org.springframework.context.annotation.Configuration
import org.springframework.graphql.execution.BatchLoaderRegistry

@Configuration
class DataLoaderRegistrationConfig(
    private val batchLoaderRegistry: BatchLoaderRegistry,
    private val bookAuthorsDataLoader: BookAuthorsDataLoader

) {
    init {
        batchLoaderRegistry.forTypePair(BookId::class.java, AuthorsWrapper::class.java)
            .registerMappedBatchLoader(bookAuthorsDataLoader)
    }
}